Solnet.Serum/MarketManager.cs

```diff
@@ -0,0 +1,874 @@
+        private readonly IList<Account> _signers;
+
+        /// <summary>
+        /// The serializer options to parse request result errors.
+        /// </summary>
+        private readonly JsonSerializerOptions _jsonSerializerOptions;
+
+        /// <summary>
+        /// The <see cref="OpenOrdersAccount"/>'s <see cref="PublicKey"/>.
+        /// </summary>
+        private PublicKey _openOrdersAccount;
+
+        /// <summary>
+        /// The decimals of the base token for the current <see cref="Market"/>.
+        /// </summary>
+        private byte _baseDecimals;
+
+        /// <summary>
+        /// The decimals of the quote token for the current <see cref="Market"/>.
+        /// </summary>
+        private byte _quoteDecimals;
+
+        /// <summary>
+        /// The subscription object corresponding to the <see cref="Market"/>'s <see cref="EventQueue"/>.
+        /// </summary>
+        private Subscription _eventQueueSubscription;
+
+        /// <summary>
+        /// The subscription object corresponding to the <see cref="Market"/>'s <see cref="OpenOrdersAccount"/>.
+        /// </summary>
+        private Subscription _openOrdersSubscription;
+
+        /// <summary>
+        /// The subscription object corresponding to the <see cref="Market"/>'s bid <see cref="OrderBookSide"/>.
+        /// </summary>
+        private Subscription _bidSideSubscription;
+
+        /// <summary>
+        /// The subscription object corresponding to the <see cref="Market"/>'s ask <see cref="OrderBookSide"/>.
+        /// </summary>
+        private Subscription _askSideSubscription;
+
+        /// <summary>
+        /// The bid side of the <see cref="OrderBook"/>.
+        /// </summary>
+        private OrderBookSide _bidSide;
+
+        /// <summary>
+        /// The ask side of the <see cref="OrderBook"/>.
+        /// </summary>
+        private OrderBookSide _askSide;
+
+        /// <summary>
+        /// Initialize the <see cref="Market"/> manager with the given market <see cref="PublicKey"/>.
+        /// </summary>
+        /// <param name="marketAccount">The <see cref="PublicKey"/> of the <see cref="Market"/>.</param>
+        /// <param name="ownerAccount">The <see cref="PublicKey"/> of the owner account.</param>
+        /// <param name="srmAccount">The <see cref="PublicKey"/> of the serum account to use for fee discount, not used when not provided.</param>
+        /// <param name="signatureMethod">A delegate method used to request a signature for transactions crafted by the <see cref="MarketManager"/> which will submit, cancel orders, or settle funds.</param>
+        /// <param name="url">The cluster to use when not passing in a serum client instance.</param>
+        /// <param name="logger">The logger instance to use.</param>
+        /// <param name="serumClient">The serum client instance to use.</param>
+        public MarketManager(PublicKey marketAccount, PublicKey ownerAccount = null, PublicKey srmAccount = null,

```

Comment:
We have a factory, make the constructor internal

Reply:
Already done with the tests